### PR TITLE
fix: Avoid breaking text on hyphens 

### PR DESCRIPTION
**Pull request by kesara**
Fixes https://github.com/ietf-tools/xml2rfc/issues/961

### CODE REVIEW

**Comment by cabo** [comment]
Wow, this change seems to have a wide-ranging effect.
Do we really want to stop breaking on hyphens everywhere?
Note that Unicode has a non-breaking hyphen character ‑, U+2011, for where this is actually desired.

**Comment by cabo** [comment]
tests/valid/draft-template.pages.text
has a nice example for why this change is generally undesirable.

File: tests/valid/draft-template.pages.text [hunk 1]
```diff
@@ -565,10 +565,10 @@ Internet-Draft              Abbreviated Title                 March 2009
    [I-D.narten-iana-considerations-rfc2434bis]
               Alvestrand, H. T. and T. Narten, "Guidelines for Writing
               an IANA Considerations Section in RFCs", Work in Progress,
-              Internet-Draft, draft-narten-iana-considerations-
-              rfc2434bis-09, March 26, 2008,
-              <https://www.ietf.org/archive/id/draft-narten-iana-
-              considerations-rfc2434bis-09.txt>.
+              Internet-Draft,
+              draft-narten-iana-considerations-rfc2434bis-09, March 26,
+              2008, <https://www.ietf.org/archive/id/draft-narten-iana-c
```
Comment (cabo) [inline tests/valid/draft-template.pages.text:570]: See https://github.com/ietf-tools/xml2rfc/pull/965/#issuecomment-1413237933

Comment (kesara) [reply tests/valid/draft-template.pages.text:570]: Yeah, this is ugly.

**Comment by kesara** [comment]
> Wow, this change seems to have a wide-ranging effect. Do we really want to stop breaking on hyphens everywhere? Note that Unicode has a non-breaking hyphen character ‑, U+2011, for where this is actually desired.

@cabo This change was just a proposal for review. I preferred this because this affects the whole text wrapping. But as you pointed out there are undesirable effects on URLs, but code can be changed to treat URLs differently.

See https://github.com/ietf-tools/xml2rfc/issues/961#issuecomment-1413347377
We better to move this conversation to that issue until we figure out which way to go.

**Comment by rjsparks** [comment]
convert this to draft?

**Comment by kesara** [comment]
Closing this as this is not the desired solution.

**Pull request closed without merging**